weave/trace/weave_client.py

```diff
@@ -930,10 +930,7 @@ def __init__(
         self._anonymous_ops: dict[str, Op] = {}
         parallelism_main, parallelism_upload = get_parallelism_settings()
         self.future_executor = FutureExecutor(max_workers=parallelism_main)
-        if parallelism_upload:
-            self.future_executor_fastlane = FutureExecutor(
-                max_workers=parallelism_upload
-            )
+        self.future_executor_fastlane = FutureExecutor(max_workers=parallelism_upload)
```

Comment:
This is a small bug. If `parallelism_upload` is 0 (falsey), we still want to create a fast line, but it will effectively be an immediate executor